hibernate-core/src/main/java/org/hibernate/query/hql/internal/QuerySplitter.java

```diff
@@ -105,8 +105,8 @@ public static <R> SqmDeleteStatement<R>[] split(SqmDeleteStatement<R> statement)
 			return sqmDeleteStatement;
 		}
 
-		final SqmPolymorphicRootDescriptor<R> unmappedPolymorphicDescriptor = (SqmPolymorphicRootDescriptor<R>) unmappedPolymorphicReference.getReferencedPathSource();
-		final Set<EntityDomainType<? extends R>> implementors = unmappedPolymorphicDescriptor.getImplementors();
+		final SqmPolymorphicRootDescriptor<?> unmappedPolymorphicDescriptor = (SqmPolymorphicRootDescriptor<?>) unmappedPolymorphicReference.getReferencedPathSource();
+		final Set<EntityDomainType<?>> implementors = unmappedPolymorphicDescriptor.getImplementors();
```

Comment:
I wouldn’t say that we need to roll back such trivial things.

Reply:
To clarify, I would not necessarily say it is trivial.  But I do think it is questionable whether the change is obvious and self-evident.  I think given the request, this is how any of us would have implemented it.

And in fact, @beikov contributed much to the actual implementation.

Let's roll back this removal